src/Illuminate/View/Compilers/ComponentTagCompiler.php

```diff
@@ -231,6 +231,9 @@ protected function componentString(string $component, array $attributes)
         }
 
         return " @component('{$class}', '{$component}', [".$this->attributesToString($parameters, $escapeBound = false).'])
+<?php if (isset($attributes) && $constructor = (new ReflectionClass('.$class.'::class))->getConstructor()): ?>
```

Comment:
This piece of code removes from `$attributes` those keys that were specified as props in the target component constructor (Class based component)